vscode-dotnet-runtime-library/src/Acquisition/DotnetCoreAcquisitionWorker.ts

```diff
@@ -69,7 +71,7 @@ export class DotnetCoreAcquisitionWorker implements IDotnetCoreAcquisitionWorker
     constructor(protected readonly context: IAcquisitionWorkerContext, private readonly utilityContext : IUtilityContext, extensionContext : IVSCodeExtensionContext) {
         const dotnetExtension = os.platform() === 'win32' ? '.exe' : '';
         this.dotnetExecutable = `dotnet${dotnetExtension}`;
-        this.acquisitionPromises = {};
+        this.acquisitionPromises = new InProgressInstallManager();
```

Comment:
I was thinking about the lifecycle of this component - I almost think it should be created when the extension initializes, and upon creation this manager should 'clean' the data in the cache, bringing it up to the current data format. this way we can ensure that all of the data is in the new format (almost like a database migration) and that the rest of the codebase doesn't have to think about the different formats.

Reply:
I think that's a wonderful idea! I've converted the class into a singleton, which I believe is the clearest way to do this pattern.

Reply:
Hm, I cant actually think of a good way to do this. The class needs the context to access the extension id of the current requesting call and for other things. We could give the context in every single function call to the object but that does not necessarily seem cleaner to me. The data in the extension state is still stored in a single place and the lock that is held to access it is shared even amongst multiple instances of this class, so both of the concerns we have *a) not thinking about different formats and *b) protecting against bad shared state is good.